site/src/sphinx/setup-configuration.rst

```diff
@@ -211,97 +211,120 @@ Core properties
-      "webAppSessionTimeoutMillis": 604800000,
-      "gracefulShutdownTimeout": {
-        "quietPeriodMillis": 1000,
-        "timeoutMillis": 10000
-      },
+      ...
       "replication" : {
-        "method" : "ZOOKEEPER",
-        "connectionString": "zk1.example.com:2181,zk2.example.com:2181,zk3.example.com:2181",
-        "pathPrefix": "/service/centraldogma",
+        "method": "ZOOKEEPER",
+        "serverId": 1,
+        "servers": {
+          "1": {
+            "host": "replica1.example.com",
+            "quorumPort": 36463,
+            "electionPort": 36464
+          },
+          "2": {
+            "host": "replica2.example.com",
+            "quorumPort": 36463,
+            "electionPort": 36464
+          },
+          "3": {
+            "host": "replica3.example.com",
+            "quorumPort": 36463,
+            "electionPort": 36464
+          }
+        },
+        "secret": "JqJAkZ!oZ6MNx4rBpIH8M*yuVWXDULgR",
+        "additionalProperties": {},
         "timeoutMillis": null,
         "numWorkers": null,
         "maxLogCount": null,
         "minLogAgeMillis": null
       },
-      "securityEnabled": false,
-      "accessLogFormat": "common",
-      "administrators": [],
-      "caseSensitiveLoginNames": false
     }
 
 - ``method`` (string)
 
-  - the replication method. ``ZOOKEEPER`` indicates ZooKeeper-based multi-master replication.
+  - the replication method. ``ZOOKEEPER`` indicates Central Dogma will provide multi-master replication by
+    embedding Apache ZooKeeper.
 
-- ``connectionString`` (string)
+- ``serverId`` (integer)
 
-  - the ZooKeeper connection string.
+  - the unique positive integer ID of the replica. Be careful not to use a duplicate ID or not to change
+    this value after joining the cluster. If ``null`` or unspecified, the ``serverId`` is auto-detected
+    from the server list in the ``servers`` section.
 
-- ``pathPrefix`` (string)
+    .. note::
 
-  - the ZooKeeper path prefix. Central Dogma will create entries under this prefix.
-  - Be extra cautious so that two different Central Dogma clusters never use the same path prefix
-    at the same ZooKeeper cluster.
+        Internally, this value is used as the ``myid`` of the embedded ZooKeeper peer.
 
-- ``timeoutMillis`` (integer)
+- ``servers``
+
+  - a map whose key is the ``serverId`` of a replica in the cluster and whose value is a map which
+    contains the properties required to connect to each other:
+
+    - ``host`` (string)
+
+      - the host name or IP address of the replica
+
+    - ``quorumPort`` (integer)
+
+      - the TCP/IP port number which is used by ZooKeeper for reaching consensus
+
+    - ``electionPort`` (integer)
 
-  - the ZooKeeper timeout, in milliseconds. If ``null``, the default value of '1000 milliseconds' (1 second)
+      - the TCP/IP port number which is used by ZooKeeper for leader election
+
+  - It is highly recommended to have more than 3 replicas because the consensus algorithm requires
```

Comment:
nit: How about adding `odd number`?  have more than 3 and odd number replicas ...

Reply:
Fixed - added some links about why odd number of ZK servers are preferred over even number of ZK servers.